extensions/blocks/pinterest/edit.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * External dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { Placeholder, SandBox, Button, IconButton, Toolbar } from '@wordpress/components';
+import { BlockControls } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { pinType } from './utils';
+import { icon } from '.';
+import './editor.scss';
+
+class PinterestEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.hideOverlay = this.hideOverlay.bind( this );
+		this.setUrl = this.setUrl.bind( this );
+
+		this.state = {
+			editedUrl: this.props.attributes.url,
+			editingUrl: false,
+			// The interactive-related magic comes from Core's EmbedPreview component,
+			// which currently isn't exported in a way we can use.
+			interactive: false,
+		};
+	}
+
+	static getDerivedStateFromProps( nextProps, state ) {
+		if ( ! nextProps.isSelected && state.interactive ) {
+			// We only want to change this when the block is not selected, because changing it when
```

Comment:
Thank you for all the comments, that's really helpful.

Reply:
Props @notnownikki, who originally wrote this code for Core, I just copy/pasted. 🙂